include/nbl/builtin/hlsl/cpp_compat/vector.hlsl

```diff
@@ -4,7 +4,7 @@
 // stuff for C++
 #ifndef __HLSL_VERSION 
 #include <stdint.h>
-
+#include <type_traits>
```

Comment:
no you can't do that, will lead to circular reference, move your `Vector` concept to `type_traits.hlsl`